components/eventing-controller/controllers/subscriptionv2/eventmesh/test/reconciler_integration_test.go

```diff
@@ -167,6 +179,114 @@ func Test_CreateSubscription(t *testing.T) {
+					},
+					{
+						Source: reconcilertesting.EventMeshNamespaceNS,
+						Type: fmt.Sprintf("%s.%s.%s1", reconcilertesting.EventMeshPrefix,
+							reconcilertesting.ApplicationName, reconcilertesting.OrderCreatedV1Event),
+					},
+				}),
+			),
+			wantEventMeshSubCheck: true,
+			wantAPIRuleCheck:      true,
+		},
+		{
+			name: "should succeed to create subscription with empty protocol and webhook settings",
+			givenSubscriptionFunc: func(namespace string) *eventingv1alpha2.Subscription {
+				return reconcilertesting.NewSubscription(testName, namespace,
+					reconcilertesting.WithDefaultSource(),
+					reconcilertesting.WithNotCleanType(),
+					reconcilertesting.WithSinkURL(reconcilertesting.ValidSinkURL(namespace, testName)),
+				)
+			},
+			wantSubscriptionMatchers: gomega.And(
+				reconcilertesting.HaveSubscriptionActiveCondition(),
+			),
+			wantEventMeshSubMatchers: gomega.And(
+				// should have default values for protocol and webhook auth
+				eventmeshsubmatchers.HaveContentMode(emTestEnsemble.envConfig.ContentMode),
+				eventmeshsubmatchers.HaveExemptHandshake(emTestEnsemble.envConfig.ExemptHandshake),
+				eventmeshsubmatchers.HaveQoS(eventMeshtypes.Qos(emTestEnsemble.envConfig.Qos)),
+				eventmeshsubmatchers.HaveWebhookAuth(eventMeshtypes.WebhookAuth{
+					ClientID:     "foo-client-id",
+					ClientSecret: "foo-client-secret",
+					TokenURL:     emTestEnsemble.envConfig.WebhookTokenEndpoint,
+					Type:         eventMeshtypes.AuthTypeClientCredentials,
+					GrantType:    eventMeshtypes.GrantTypeClientCredentials,
+				}),
+			),
+			wantAPIRuleCheck:      true,
+			wantEventMeshSubCheck: true,
+		},
+		{
+			name: "should succeed to create subscription with EXACT type matching",
+			givenSubscriptionFunc: func(namespace string) *eventingv1alpha2.Subscription {
+				return reconcilertesting.NewSubscription(testName, namespace,
+					reconcilertesting.WithExactTypeMatching(),
+					reconcilertesting.WithEventMeshExactType(),
+					reconcilertesting.WithSinkURL(reconcilertesting.ValidSinkURL(namespace, testName)),
+				)
+			},
+			wantSubscriptionMatchers: gomega.And(
+				reconcilertesting.HaveSubscriptionFinalizer(eventingv1alpha2.Finalizer),
+				reconcilertesting.HaveSubscriptionActiveCondition(),
+				reconcilertesting.HaveCleanEventTypes([]eventingv1alpha2.EventType{
+					{
+						OriginalType: reconcilertesting.EventMeshExactType,
+						CleanType:    reconcilertesting.EventMeshExactType,
+					},
+				}),
+			),
+			wantEventMeshSubMatchers: gomega.And(
+				eventmeshsubmatchers.HaveEvents(eventMeshtypes.Events{
+					{
+						Source: reconcilertesting.EventMeshNamespaceNS,
+						Type:   reconcilertesting.EventMeshExactType,
+					},
+				}),
+			),
+			wantEventMeshSubCheck:    true,
+			wantAPIRuleCheck:         true,
+			wantSubCreatedEventCheck: true,
+			wantSubActiveEventCheck:  true,
+		},
+		{
+			name: "should mark a non-cleaned Subscription as ready",
```

Comment:
(nit) The `non-cleaned` gives the impression that something went wrong while cleaning-up the subscription. Maybe `not clean` is more convenient here.